documentation/onboard-dpg-in-sdkautomation/Integrate-dpg-and-apiview-in-sdk-automation-pipeline.md

```diff
@@ -0,0 +1,31 @@
+# Integrate DPG and ApiView in SDK Automation Pipeline
+This document is targeting for everyone who wants to know all the implementation details of SDK automation pipeline.
+It describes the details of integrating DPG and ApiView in SDK Automation Pipeline.
+
+Before go through this document, please go through [Service Onboard DPG with Swagger CI Pipeline](README.md) to be familiar with the user experience in service team side. Also, you need to be familiar with [SDK Automation Pipeline Framework](../sdkautomation/README.md).
+
+# WorkFlow
+![integrate-dpg-and-apiview](integrate-dpg-and-apiview.png)
```

Comment:
I recommend you checkout [mermaid.js](https://mermaid-js.github.io/mermaid/#/). GitHub supports mermaid.js natively so there is no need to check in .png files, which means a year from now no one has to hunt down the source and tools used to create the .png.

Reply:
I maybe not a good mermaid user. When I try to migrate the workflow to mermaid, the flowchart rendered by mermaid is not easy for read. The following is the flowchart:
```mermaid
flowchart TD
   serviceTeam(Service Team) --> swaggerPR(Swagger PR)
   swaggerPR --> choose1{Have you added a comment containing autorest configuration\n in spec PR and run command `/azp run`?}
   choose1 -->|Yes| generate1(Use the information in comment to generate sdk and ApiView)
   generate1 --> choose2{Is API Definition Approvied?}
   choose2 -->|Yes| END(End)
   choose2 -->|No| change(Add/Update the comment that contains autorest configuraiton, and then re-run pipeline by command `/azp run`)
   change --> generate1
   choose1 -->|No| choose3{Is there autorest configuration file in sdk repo?}
   choose3 -->|No| END
   choose3 -->|Yes| generate2(Generate sdk by using autorest configuration in sdk repo)
   generate2 --> choose2
```